### DISCUSSION TITLE
Jit export for VAD model

### DISCUSSION THREAD

**Discussion by samr66** [speechbrain/speechbrain#2019] [Q&A] [answered]
Hi, is there a turtorial for exporting VAD model (or similar) to jit? If that is possible to do?

**Comment by Adel-Moumen** [comment]
Hey @fpaissan could you please help him? Many thanks. :-)

**Comment by fpaissan** [comment, marked as answer]
Hey @samr66,

We don't have a tutorial on how to export the VAD model to ONNX, but it's currently being developed. 
However, please note that ONNX still does not support FFT (https://github.com/pytorch/pytorch/issues/81075#issuecomment-1539099760), so the model is not exportable as-is but should be slightly modified to remove the feature extraction from the model.

Hope this was useful.. let me know if you have any other questions.. :)

**Comment by AdityaYadavalli1** [comment]
Hey @fpaissan! 

Really looking forward to the tutorial! Approximately by when would this be ready?

Currently, we are trying to export a jit version of the VAD model so that we can have a more portable version of the model (without speechbrain dependency, torch dependency is ok for now).

I tried doing this but the outputs of the exported model and the original model differ, which makes me suspect that I am making a mistake somewhere (happy to share my code with you, if that helps!). Could you help me with a possible pseudo-code that I should be following?

**Comment by fpaissan** [comment]
Sure, @AdityaYadavalli1, if you have a code snippet I could use to reproduce your problem, I might be able to help debug it.

The tutorial should be a matter of a couple of months. I'll ping you as soon as we release it.

Cheers

**Comment by AdityaYadavalli1** [comment]
Thank you so much @fpaissan! I will share the notebook with you shortly!

**Comment by emmanuelrobles** [comment]
Hey, is there any news on the documentation on how to export the model, I am trying to export the G2P model and I am running into all kinds of issues, thanks!
